Stratis.Bitcoin/Features/MemoryPool/MempoolManager.cs

```diff
@@ -9,23 +9,49 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Memory pool scheduler.
+    /// </summary>
     public class MempoolScheduler : AsyncLock
```

Comment:
I will rename it. What do you think about pulling it out into a separate .cs file? At some places I've worked they had a rule about only one class per source file. Personally I think it's best to be in a separate source file if it's consumed anywhere outside of that source file.